cypress/integration/sales/sales_order_to_invoice_spec.js

```diff
@@ -0,0 +1,144 @@
+import { salesInvoices } from '../../page_objects/sales_invoices';
+
+describe('New sales order test', function() {
+  const timestamp = new Date().getTime();
+
+  const poReference = `Sales Order-to-Invoice Test ${timestamp}`;
+  const productName = `Sales Order-to-Invoice Test ${timestamp}`;
+  const customerName = `Sales Order-to-Invoice Test ${timestamp}`;
+
+  before(function() {
+    const productValue = `sales_order_to_invoice_test ${timestamp}`;
+    const productCategoryName = `ProductCategoryName ${timestamp}`;
+    const productCategoryValue = `ProductNameValue ${timestamp}`;
+
+    cy.fixture('product/simple_productCategory.json').then(productCategoryJson => {
+      Object.assign(new ProductCategory(), productCategoryJson)
+        .setName(productCategoryName)
+        .setValue(productCategoryValue)
+        .apply();
+    });
+    cy.fixture('product/simple_product.json').then(productJson => {
+      Object.assign(new Product(), productJson)
+        .setName(productName)
+        .setValue(productValue)
+        .setProductType('Service')
+        .setProductCategory(productCategoryValue + '_' + productCategoryName)
+        .apply();
+    });
+    cy.fixture('sales/simple_customer.json').then(customerJson => {
+      Object.assign(new BPartner(), customerJson)
+        .setName(customerName)
+        .apply();
+    });
+  });
+
+  describe('Create a new sales order', function() {
+    it('Create new sales order header', function() {
+      cy.visitWindow(salesOrders.windowId, 'NEW');
+
+      cy.writeIntoLookupListField('C_BPartner_ID', customerName, customerName);
+
+      cy.get('.header-breadcrumb-sitename').should('not.contain', '<');
+
+      cy.writeIntoStringField('POReference', poReference, false /*modal*/, null /*rewriteUrl*/, true /*noRequest*/);
+      cy.get('.indicator-pending').should('not.exist');
+    });
+
+    it('Add new product via Batch Entry', function() {
+      const addNewText = Cypress.messages.window.batchEntry.caption;
+
+      cy.get('.tabs-wrapper .form-flex-align .btn')
+        .contains(addNewText)
+        .should('exist')
+        .click();
+
+      cy.get('.quick-input-container').should('exist');
+
+      cy.writeIntoLookupListField('M_Product_ID', productName, productName);
+
+      const aliasName = `addProduct-${timestamp}`;
+      const patchUrlPattern = '/rest/api/window/.*$';
+      cy.server();
+      cy.route('GET', new RegExp(patchUrlPattern)).as(aliasName);
+
+      cy.get('.form-field-Qty')
+        .find('input')
+        .type('1{enter}');
+
+      cy.wait(`@${aliasName}`);
+    });
+
+    it('Complete sales order', function() {
+      cy.processDocument('Complete', 'Completed');
+    });
+  });
+
+  describe('create an invoice', function() {
+    it("Zoom to the sales order's invoice candidate", function() {
+      cy.wait(5000); // wait a bit for the invoice candidate(s) to be created, just like the user would
```

Comment:
Any ideas if we can get around this without deliberately waiting ?

Reply:
@siemiatj currently not..we might make the backend fire a notification for this as well..but very often the user is not interested in this particular notification..so i think to get it done nicely, we need to do a lot more thinking..
oor i wonder if we can have "invisible" notifications..which we can access with cypress but which are generally not shown to the user..this would also scratch another bad itch that i have about making cache invalidations observable..